ipfs-cluster/trigger-server/triggerserver.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+    "log"
+    "net/http"
+    "os/exec"
+    "time"
+    "fmt"
+    "io/ioutil"
+    "math/rand"
+    "strconv"
+    "os"
+    "github.com/ipfs/go-ipfs-api"
+    "bytes"
+)
+
+const CONFIG_PATH = "/etc/trigger-server"
+// HTTP API listen port
+const PORT = 8082
+var RUNNER_PATH = os.Getenv("GOPATH") + "src/github.com/ipfs/kubernetes-ipfs/ipfs-cluster/"
+// Runner takes two args
+const RUNNER_NUM_NODES = "5"
+const RUNNER_NUM_PINS = "5"
+
+// Secret is set by main
+var secret = ""
+
+// IPFS Shell
+var sh = shell.NewShell("localhost:5001")
+
+func run(rw http.ResponseWriter, req *http.Request) {
+    if req.Method == "POST" {
+	if req.Header.Get("X-Auth") != secret {
+		log.Println("User failed authentication")
+		fmt.Fprintf(rw, "Invalid authentication\n")
+	} else {
+		// This command takes a while, so we'll hang out here while it runs
+		cmd := exec.Command("bash", "runner.sh", RUNNER_NUM_NODES, RUNNER_NUM_PINS)
+		cmd.Dir = RUNNER_PATH
+		out, err := cmd.CombinedOutput()
```

Comment:
find a way to stream the output of the command as it happens. Seems https://golang.org/pkg/os/exec/#Cmd.StdinPipe might help. May want to run with a command context so you can cancel the command if you cannot write anymore.

Also, should make sure the deployment is cleaned up from the cluster after the tests have run so they can run again without finding anything dirty.